### PR TITLE
feature: hide current release when LTS release is more recent

### PR DESCRIPTION
**Pull request by MylesBorins**
This is an edge case we hit today. The latest Current release is 12.12.0,
the latest LTS release is 12.13.0. This leads to a confusing landing page.

This adds an additional semver comparison and hides the current release
from the main page on days like today when it shouldn't be shown.

Refs: https://github.com/nodejs/Release/issues/495

### CODE REVIEW

File: layouts/index.hbs [hunk 1]
```diff
@@ -42,6 +42,7 @@
         </div>
 
         {{#if project.latestVersions.current.node}}
+        {{#if project.latestVersions.current.hideCurrent}}
```
Comment (ljharb) [inline layouts/index.hbs]: do you mean this to render when `hideCurrent` is true? or the inverse, when hideCurrent is false?

Comment (MylesBorins) [reply layouts/index.hbs]: fixed.

**Comment by ljharb** [comment]
LGTM!

**Review by nschonni — APPROVED** [review]

**Review by watson — APPROVED** [review]

**Comment by XhmikosR** [comment]
@MylesBorins shouldn't we do something similar for https://nodejs.org/en/download/ ?

**Pull request merged**